### PR TITLE
GH-36641: [C++] Remove reference to acero from non-acero file

### LINKED ISSUE #36641: [C++] util_avx2.cc has been moved to compute but still #include acero headers
### Describe the bug, including details regarding any error messages, version, and platform.

It looks like https://github.com/apache/arrow/issues/35115 moved util_avx2.cc from acero/ to compute/ but forgot to update `#include` lines. This was discovered while building Apache Arrow with a sandboxed build system where the entire acero/ is not visible if ARROW_ACERO=OFF.

### Component(s)

C++

### PR DESCRIPTION
**Pull request by westonpace**
### Rationale for this change

Files in modules which do not depend on the acero module should not reference files inside the acero module.

### What changes are included in this PR?

There were no changes to the body of any functions.  I simply moved functions around so that the acero include was no longer needed.  There were some conflicts that arose between the class `bit_util` and the namespace `bit_util` and so I got rid of the class in favor of the namespace as that is more similar to how we handle `bit_util` elsewhere.

### Are these changes tested?

Sort of.  I would like to add an AVX2 CI system as well.  I'm not confident any of the CI builds are building with AVX2 enabled.  Also, even if we have an AVX2 CI system it would not have caught this issue since the code was only needed definitions from the acero header and was not relying on any actual compiled symbols.  However, I think setting up tests to catch this sort of invalid include are beyond the scope of this PR.

### Are there any user-facing changes?

No.
* Closes: #36641

### CODE REVIEW

File: cpp/src/arrow/compute/util.h [hunk 1]
```diff
@@ -139,68 +139,54 @@ class TempVectorHolder {
...
+                                     uint8_t* bytes);
+ARROW_EXPORT void bytes_to_bits_avx2(const int num_bits, const uint8_t* bytes,
+                                     uint8_t* bits);
+ARROW_EXPORT bool are_all_bytes_zero_avx2(const uint8_t* bytes, uint32_t num_bytes);
+}  // namespace avx2
+
 #endif
-};
+
+};  // namespace bit_util
```
Comment (pitrou) [inline cpp/src/arrow/compute/util.h:189]: Spurious semicolon here?

File: cpp/src/arrow/compute/util_avx2.cc [hunk 2]
```diff
@@ -16,30 +16,19 @@
 // under the License.
 
 #include <immintrin.h>
+#include <cstring>
```
Comment (pitrou) [inline cpp/src/arrow/compute/util_avx2.cc:19]: Keep this alphabetically-ordered?

File: cpp/src/arrow/compute/util.h (hunk 1 above)
Comment (pitrou) [inline cpp/src/arrow/compute/util.h:189]:
```suggestion
}  // namespace bit_util
```

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
> Sort of. I would like to add an AVX2 CI system as well. I'm not confident any of the CI builds are building with AVX2 enabled.

Probably not, though it should be easy to enable one of them. AVX2 is probably available on most server x86 CPUs nowadays?

**Comment by pitrou** [comment]
I opened https://github.com/apache/arrow/pull/36662 to exercise compile-time AVX2 on CI.

**Comment by pitrou** [comment]
@raulcd This would be nice to have in 13.0.0 if not too late.

**Comment by raulcd** [comment]
Thanks, yes, I will cherry-pick

**Pull request merged**